examples/ui/ui.rs

```diff
@@ -18,288 +18,303 @@ fn setup(mut commands: Commands, asset_server: Res<AssetServer>) {
 
     // root node
     commands
-        .spawn_bundle(NodeBundle {
-            style: Style {
-                size: Size::new(Val::Percent(100.0), Val::Percent(100.0)),
-                justify_content: JustifyContent::SpaceBetween,
-                ..Default::default()
-            },
-            color: Color::NONE.into(),
-            ..Default::default()
-        })
+        .spawn_bundle(root_element())
```

Comment:
You could reduce the nesting by reusing the root elements `EntityCommands` instead of having a root level `with_children`,

```rust
let root = root.spawn_bundle(root_element());

 root.spawn_bundle(left_vertical_fill_border())
    .with_children(|parent| {
       ...
    });

root.spawn_bundle(right_vertical_fill())
    .with_children(|parent| {
        ...
    });
```
etc